cmd/cloud-controller-manager/.import-restrictions

```diff
@@ -1,6 +1,8 @@
 rules:
   - selectorRegexp: k8s[.]io/kubernetes
     allowedPrefixes:
+      - k8s.io/kubernetes/cmd/kube-controller-manager/app/options
```

Comment:
When we talked to @liggitt about these changes, one of the things which was brought up was clearly delineating the sample (what we expect cloud providers to copy) from the example (showing how to extend the sample to do things like add new controllers). This seems to emphasize the importance of separating the sample from the example. This *only* belongs in the example and should not be in the sample.

Reply:
Seems like we should have a README.md file in cmd/cloud-controller-manager which clearly states that this directory is an example (POC) of using the CCM extension mechanisms. Further that new cloud providers should copy the CCM from k8s.io/cloud-provider/sample/basic_main.go. Possibly even basic instructions for doing this.

Reply:
Added README